Sync docs automation (#3022)
kyma-project/kyma-environment-broker: docs/contributor/04-10-workflows.md

4. Uploads the downloaded logs as assets to the current GitHub release
5. Uploads the downloaded logs to a GCP bucket

## Sync KEB Docs Structure Workflow

> The [`sync-docs-toc`](/.github/workflows/sync-docs-toc.yml) workflow automatically opens a PR to the `product-kyma-runtime` repository whenever the structure of KEB's `docs/` directory changes — specifically when a published document is added or renamed. It is triggered after every successful [Promote KEB to DEV](#promote-keb-to-dev-workflow) workflow run.

Reviewer: That's not quite true. Let's be more accurate. Would this version work for you?

The [`sync-docs-toc`](/.github/workflows/sync-docs-toc.yml) workflow automatically opens a PR to the `product-kyma-runtime` repository whenever the `<!--{"metadata":{"publish":true}}-->` metadata is added to a file in KEB's `docs/` directory. This occurs when a new document is added, or an existing one is renamed, or the metadata is changed from `"publish":false` to `"publish":true`.
These documents are published to the Restricted Markets documentation. Pure content edits to existing files require no action — `product-kyma-runtime` pulls the latest content automatically using a `fileTree` reference in its `manifest.yaml`.
The workflow is triggered after every successful [Promote KEB to DEV](#promote-keb-to-dev-workflow) workflow run.

Author: sure, looks good

> Only files marked with `<!--{"metadata":{"publish":true}}-->` metadata are considered. These are documents published to the Restricted Markets documentation. Pure content edits to existing files require no action — `product-kyma-runtime` pulls the latest content automatically via a `fileTree` reference in its `manifest.yaml`.

Reviewer: ```suggestion
```

The workflow performs the following steps:

1. Resolves the current and previous KEB release tags to determine the diff range
2. Detects added or renamed `.md` files in `docs/` between the two releases, filtered by `publish:true` metadata
3. Fetches the corresponding chart version from `management-plane-charts` (falls back to searching merged PRs if the release branch was already deleted)

> 4. Updates `docs/toc.yaml` in `product-kyma-runtime` using the [`scripts/python/update_toc.py`](../../scripts/python/update_toc.py) script, which inserts new entries in numeric-prefix order and renames existing ones

Reviewer: ```suggestion
4. Updates `docs/toc.yaml` in `product-kyma-runtime` using the [`scripts/python/update_toc.py`](../../scripts/python/update_toc.py) script, which inserts new entries in numeric-prefix order and updates the renamed ones
```

4. Uploads the downloaded logs as assets to the current GitHub release
5. Uploads the downloaded logs to a GCP bucket

## Sync KEB Docs Structure Workflow

The [`sync-docs-toc`](/.github/workflows/sync-docs-toc.yml) workflow automatically opens a PR to the `product-kyma-runtime` repository whenever the `<!--{"metadata":{"publish":true}}-->` metadata is added to a file in KEB's `docs/` directory. This occurs when a new document is added, or an existing one is renamed, or the metadata is changed from `"publish":false` to `"publish":true`. 

> These documents are published to the Restricted Markets documentation. Pure content edits to existing files require no action — `product-kyma-runtime` pulls the latest content automatically using a `fileTree` reference in its `manifest.yaml`.

Reviewer: ```suggestion
These documents are published to the Restricted Markets documentation. Content edits to existing files require no action — `product-kyma-runtime` pulls the latest content automatically using a `fileTree` reference in its `manifest.yaml`.
```

The workflow is triggered after every successful [Promote KEB to DEV](#promote-keb-to-dev-workflow) workflow run.

> 

Reviewer: ```suggestion
```
